rotkehlchen/chain/ethereum/modules/uniswap/utils.py

```diff
@@ -253,3 +273,319 @@ def historical_uniswap_v2_lp_price(
+                decimal_1=entry[1]['decimals'],
+            )
+            for entry in zip(tokens_a, tokens_b, positions)
+        ]
+        amounts_0 = [
+            calculate_amount_0(
+                tick=entry[0][1],
+                tick_bottom=entry[1][5],
+                tick_top=entry[1][6],
+                liquidity=entry[1][7],
+                decimals=entry[2]['decimals'],
+            )
+            for entry in zip(slots_0, positions, tokens_a)
+        ]
+        amounts_1 = [
+            calculate_amount_1(
+                tick=entry[0][1],
+                tick_bottom=entry[1][5],
+                tick_top=entry[1][6],
+                liquidity=entry[1][7],
+                decimals=entry[2]['decimals'],
+            )
+            for entry in zip(slots_0, positions, tokens_b)
+        ]
+        total_tokens_in_pools = []
+        if premium:
+            try:
+                liquidity_in_pools_multicall = multicall_2(
+                    ethereum=ethereum,
+                    require_success=False,
+                    calls=[
+                        (entry[0], entry[1].encode('liquidity'))
+                        for entry in zip(pool_addresses, pool_contracts)
+                    ],
+                )
+            except RemoteError as e:
+                log.error(UNISWAP_V3_ERROR_MSG.format('pool contract liquidity', str(e)))
+                continue
+            liquidity_in_pools = [
+                (entry[0].decode(entry[1][1], 'liquidity')[0])
+                for entry in zip(pool_contracts, liquidity_in_pools_multicall) if entry[1][0] is True  # noqa: 501
+            ]
+            for entry in zip(slots_0, liquidity_in_pools, positions, tokens_a, tokens_b):
+                total_tokens_in_pools.append(
+                    calculate_total_amounts_of_tokens(
+                        liquidity=entry[1],
+                        tick=entry[0][1],
+                        fee=entry[2][4],
+                        decimal_0=entry[3]['decimals'],
+                        decimal_1=entry[4]['decimals'],
+                    ),
+                )
+        for entry in itertools.zip_longest(
+            tokens_ids,
+            pool_addresses,
+            positions,
+            price_ranges,
+            tokens_a,
+            tokens_b,
+            amounts_0,
+            amounts_1,
+            total_tokens_in_pools,
+            fillvalue=(None, None),
+        ):
+            if FVal(entry[6]) > ZERO or FVal(entry[7]) > ZERO:  # type: ignore
+                entry[4].update({'amount': entry[6], 'address': entry[2][2], 'total_amount': entry[8][0]})  # type: ignore  # noqa: 501
+                entry[5].update({'amount': entry[7], 'address': entry[2][3], 'total_amount': entry[8][1]})  # type: ignore  # noqa: 501
+                balances.append(_decode_v3_result(userdb, entry, known_assets, unknown_assets))
+    return balances
+
+
+def compute_pool_address(
+    factory_address: ChecksumEthAddress,
```

Comment:
Why is factory_address even an argument here? It's given as a constant at the one place `compute_pool_address` is used.